src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Workspace.cs

```diff
@@ -218,10 +224,13 @@ async Task ProcessInitialEvaluationDataAsync(CancellationToken cancellationToken
 
             if (string.IsNullOrEmpty(languageName) || string.IsNullOrEmpty(binOutputPath) || string.IsNullOrEmpty(projectFilePath))
             {
-                // Insufficient data to initialize the language service.
-                _state = WorkspaceState.Failed;
+                // It's reasonable for a project to exist that doesn't contain enough data to initialize the language
+                // service. For example, the ".ilproj" project is not uncommon and hits this code path.
+                // In such cases, we don't want to throw here as that would fault the project. Instead, we
+                // dispose the workspace tears down the dataflow subscription and clean everything up gracefully.
+                Dispose();
```

Comment:
Would it be possible to log a message about this somewhere? I worry about misconfigured .csproj/.vbproj/.fsproj projects; previously we would have thrown an exception, but now everything might appear to work and you just don't get Intellisense.

Reply:
> previously we would have thrown an exception

In 17.3 (prior to the perf work on the language service integration) there was no exception and no log message, so this PR just restores the old behaviour. The exception was a regression in the 17.4 previews until now.

Logging a message might make sense in some cases, however in other cases it might not. The customer who discovered this bug in 17.4 preview 2.1 was using an `.ilproj` which doesn't fire up Roslyn. Logging a message for such projects could be noisy, depending upon how we do it.

Can you think of a good way to balance these?